### PR TITLE
[docs] Update the upgrade guide

### PR DESCRIPTION
**Pull request by Jennifer88huang**
### Motivation
The original PR https://github.com/apache/pulsar/pull/4770 is merged before I could give some feedback, so I just create a new pull request.

### Modifications
1. Refine the structure, remove some unnecessary info currently. Thus it is displayed to users as a complete guide. 
2. Refine the language part, make it clear and concise.
3. Though "upgrade ZooKeeper" is optional, could we add information for it? So users can decide whether they want to upgrade ZooKeeper. When they upgrade ZooKeeper, they know how to upgrade.

### CODE REVIEW

**Comment by Jennifer88huang** [comment]
@sijie Could you please add the content for "Upgrade ZooKeeper (optional)" section? Thank you.

**Comment by Jennifer88huang** [comment]
Is there any update on this PR? Any issue, feel free to comment. Thank you.

File: site2/docs/administration-upgrade.md [hunk 1]
```diff
@@ -1,182 +1,130 @@
...
-  2. Rolling Upgrade: rollout the upgraded version to all proxies in the cluster once you determined a version is safe after canary.
+3. Upgrade brokers
+- Canary test: test an upgraded version in one or a small set of brokers.
+- Rolling upgrade: rollout the upgraded version to all brokers in the cluster after you have determined a version is safe after canary.
+4. Upgrade proxies
+- Canary test: test an upgraded version in one or a small set of proxies.
+- Rolling upgrade: rollout the upgraded version to all proxies in the cluster after you have determined a version is safe after canary.
 
-## Upgrade Bookies
+## Upgrade ZooKeeper (optional)
```
Comment (sijie) [inline site2/docs/administration-upgrade.md]:
```
While upgrading bookies, you can do canary test first, and then upgrade all zookeeper in the cluster.

### Canary test

You can test an upgraded version in one of zookeeper servers before upgrading all zookeeper servers in your cluster.

To upgrade zookeeper server to a new version, complete the following steps:

1. Stop a zookeeper server.
2. Upgrade the binary and configuration files.
3. Start the zookeeper server with the new binary.
4. Use `pulsar zookeeper-shell` to connect to the newly upgraded zookeeper server and run a few commands to verify if it is working as expected.
5. Let the zookeeper server run for a few days, observe and make sure the zookeeper cluster is running fine without any errors.

#### Canary rollback

If issues occur during canary test, you can shut down the problematic zookeeper node, revert the binary and configuration, and restart the zookeeper with the reverted binary.

### Upgrade all zookeeper servers

After canary test to upgrade one zookeeper in your cluster, you can upgrade all zookeeper servers in your cluster. 

You can upgrade all the zookeeper servers one by one by following the steps mentioned in canary test.
```

**Comment by Jennifer88huang** [comment]
@sijie Thank you very much for your content. I've added "Upgrade ZooKeeper" section. PTAL.

**Pull request merged**